pkg/cgroups/cgroupblkio.go

```diff
@@ -0,0 +1,188 @@
+	return OciBlockIOParameters{
+		Weight:                  -1,
+		LeafWeight:              -1,
+		WeightDevice:            nil,
+		ThrottleReadBpsDevice:   nil,
+		ThrottleWriteBpsDevice:  nil,
+		ThrottleReadIOPSDevice:  nil,
+		ThrottleWriteIOPSDevice: nil,
+	}
+}
+
+// NewOciWeightDeviceParametes creates new OciWeightDeviceParameters instance.
+func NewOciWeightDeviceParametes() OciWeightDeviceParameters {
+	return OciWeightDeviceParameters{
+		Major:      -1,
+		Minor:      -1,
+		Weight:     -1,
+		LeafWeight: -1,
+	}
+}
+
+// NewOciWeightDeviceParametes creates new OciRateDeviceParameters instance.
+func NewOciRateDeviceParameters() OciRateDeviceParameters {
+	return OciRateDeviceParameters{
+		Major: -1,
+		Minor: -1,
+		Rate:  -1,
+	}
+}
+
+// GetBlkioDir returns the cgroups blkio controller directory.
+func GetBlkioDir() string {
+	return blkioCgroupDir
+}
+
+// SetBlkioParameters writes OCI BlockIO parameters to files in cgroups blkio contoller directory.
+func SetBlkioParameters(cgroupsDir string, blockIO OciBlockIOParameters) error {
+	logger.Debug("configuring cgroups blkio controller in directory %#v with parameters %+v", cgroupsDir, blockIO)
+	var errors = []error{}
+
+	if blockIO.Weight >= 0 {
+		weightString := strconv.FormatInt(blockIO.Weight, 10)
+		errors = append(errors, writeToFilesInDir(cgroupsDir, blkioWeightFiles, weightString)...)
+	}
+
+	if blockIO.LeafWeight >= 0 {
+		leafWeightString := strconv.FormatInt(blockIO.LeafWeight, 10)
+		errors = append(errors, writeToFilesInDir(cgroupsDir, blkioLeafWeightFiles, leafWeightString)...)
+	}
+
+	for _, weightDevice := range blockIO.WeightDevice {
+		weightDeviceString := fmt.Sprintf("%d:%d %d", weightDevice.Major, weightDevice.Minor, weightDevice.Weight)
+		errors = append(errors, writeToFilesInDir(cgroupsDir, blkioWeightDeviceFiles, weightDeviceString)...)
+	}
+
+	for _, rateDevice := range blockIO.ThrottleReadBpsDevice {
+		rateDeviceString := fmt.Sprintf("%d:%d %d", rateDevice.Major, rateDevice.Minor, rateDevice.Rate)
+		errors = append(errors, writeToFilesInDir(cgroupsDir, blkioThrottleReadBpsFiles, rateDeviceString)...)
+	}
+
+	for _, rateDevice := range blockIO.ThrottleWriteBpsDevice {
+		rateDeviceString := fmt.Sprintf("%d:%d %d", rateDevice.Major, rateDevice.Minor, rateDevice.Rate)
+		errors = append(errors, writeToFilesInDir(cgroupsDir, blkioThrottleWriteBpsFiles, rateDeviceString)...)
+	}
+
+	for _, rateDevice := range blockIO.ThrottleReadIOPSDevice {
+		rateDeviceString := fmt.Sprintf("%d:%d %d", rateDevice.Major, rateDevice.Minor, rateDevice.Rate)
+		errors = append(errors, writeToFilesInDir(cgroupsDir, blkioThrottleReadIOPSFiles, rateDeviceString)...)
+	}
+
+	for _, rateDevice := range blockIO.ThrottleWriteIOPSDevice {
+		rateDeviceString := fmt.Sprintf("%d:%d %d", rateDevice.Major, rateDevice.Minor, rateDevice.Rate)
+		errors = append(errors, writeToFilesInDir(cgroupsDir, blkioThrottleWriteIOPSFiles, rateDeviceString)...)
+	}
+
+	err := utils.JoinErrors(errors)
```

Comment:
nitpick: why not just `return util.JoinErrors(errors)` ?

Reply:
Converted to multierror and implemented as you suggest.